rfc/draft-oberstet-hybi-tavendo-wamp-00.md

```diff
@@ -41,91 +41,84 @@ This document defines the Web Application Messaging Protocol (WAMP). WAMP is a r
 
-This document first describes a Basic Profile for WAMP in its entirety before describing an Advanced Profile which extends the functionality of WAMP.
+### Basic and Advanced Profiles
 
-The separation in Basic and Advanced Profile is intended to extend the reach of the protocol. 
+This document first describes a Basic Profile for WAMP in its entirety, before describing an Advanced Profile which extends the basic functionality of WAMP.
 
-It allows implementations to start out with a minimal but operable and useful set of functionality, and to expand their functionality from there.
+The separation into Basic and Advanced Profiles is intended to extend the reach of the protocol. It allows implementations to start out with a minimal, yet operable and useful set of features, and to expand that set from there. It also allows implementations that are tailored for resource-constrained environments, where larger feature sets would not be possible. Here implementers can weigh between resource constraints and functionality requirements, then implement an optimal feature set for the circumstances.
 
-It also allows implementations in resource-constrained environments where an implementation of a larger set of functionality would not be possible. Here implementers can weigh between resource constraints and functionality requirements and implement an optimal set for the circumstances.
-
-Feature announcement is used for Advanced Profile features, so that different implementations can adjust their interactions to fit the commonly supported feature set.
+Advanced Profile features are announced during session establishment, so that different implementations can adjust their interactions to fit the commonly supported feature set.
 
 
 ### Application Code
 
-WAMP is designed for application code to run inside *Clients*, i.e. *Peers* of the roles *Callee*, *Caller*, *Publisher*, and *Subscriber*.
+WAMP is designed for application code to run within *Clients*, i.e. *Peers* having the roles *Callee*, *Caller*, *Publisher*, and *Subscriber*.
 
 *Routers*, i.e. *Peers* of the roles *Brokers* and *Dealers* are responsible for **generic call and event routing** and do not run application code.
 
-This allows to transparently exchange *Broker* and *Dealer* implementations without affecting the application and to distribute and deploy application components flexibly.
+This allows the transparent exchange of *Broker* and *Dealer* implementations without affecting the application and to distribute and deploy application components flexibly.
```

Comment:
"This allows to" sounds bad to ear. See http://english.stackexchange.com/questions/85069.